tests/commons/src/main/java/org/codice/ddf/admin/comp/test/AbstractComponentTest.java

```diff
@@ -0,0 +1,177 @@
+                karafDistributionConfiguration().frameworkUrl(maven().groupId("org.codice.ddf")
+                        .artifactId("kernel")
+                        .versionAsInProject()
+                        .type("zip"))
+                        .unpackDirectory(new File("target/exam"))
+                        .useDeployFolder(false),
+                cleanCaches());
+    }
+
+    public List<Option> configurableSettings() throws Exception {
+        return Arrays.asList(keepRuntimeFolder(),
+                logLevel(LogLevelOption.LogLevel.INFO),
+
+                // TODO: tbatie - 8/15/17 - Change these to dynamic port once refactored
+                editConfigurationFilePut("etc/system.properties", "ddf.home", "${karaf.home}"),
+                editConfigurationFilePut("etc/system.properties",
+                        "org.codice.ddf.system.httpsPort",
+                        "9993"),
+                editConfigurationFilePut("etc/system.properties",
+                        "org.codice.ddf.system.httpPort",
+                        "9994"),
+                editConfigurationFilePut("etc/system.properties",
+                        "org.codice.ddf.catalog.ftp.port",
+                        "9995"),
+                editConfigurationFilePut("etc/org.apache.karaf.management.cfg",
+                        "rmiRegistryPort",
+                        "20001"),
+                editConfigurationFilePut("etc/org.apache.karaf.management.cfg",
+                        "rmiServerPort",
+                        "20002"),
+                editConfigurationFilePut("etc/org.apache.karaf.management.cfg",
+                        "rmiServerPort",
+                        "20002"),
+                editConfigurationFilePut("etc/org.apache.karaf.features.cfg",
+                        "featuresBootAsynchronous",
+                        "true"));
+    }
+
+    @BeforeExam
+    public void beforeExam() throws Exception {
+        adminConfig = new AdminConfig(configurationAdmin);
+        serviceManager = (ServiceManager) Proxy.newProxyInstance(getClass().getClassLoader(),
+                ServiceManagerImpl.class.getInterfaces(),
+                new ServiceManagerProxy(new ServiceManagerImpl(metatypeService, adminConfig)));
+
+        securityPolicyConfigurator = new SecurityPolicyConfigurator(new ServiceManagerImpl(
+                metatypeService,
+                adminConfig), configurationAdmin);
+        startFeatures();
+    }
+
+    protected List<Option> configureVmOptions() {
+        return Arrays.asList(vmOption("-Xmx4096M"),
+                // avoid integration tests stealing focus on OS X
+                vmOption("-Djava.awt.headless=true"), vmOption("-Dfile.encoding=UTF8"));
+    }
+
+    public void startFeatures() throws Exception {
+        features().stream()
+                .filter(f -> !f.isBootFeature() && f.getFeatureName() != null)
+                .map(Feature::getFeatureName)
+                .parallel()
```

Comment:
✏️ It doesn't matter that much whether you add the call to `parallel` here or above, but from a review perspective it's better to put it immediately after `.stream()` so people can see that this invocation is parallelized.